### PR TITLE
GP2HUB-89 Add first section fields to research output

### PR DESCRIPTION
**Pull request by gabiayako**
Jira ticket: https://asaphub.atlassian.net/browse/GP2HUB-89

---

This PR
- Adds a migration to add the fields description, gp2Supported and sharingStatus to the Outputs schema in Contentful
- Adds description, gp2Supported and sharingStatus to the output query and the data provider response
- Adds the inputs for these description, gp2Supported, sharingStatus and publishDate to the OutputForm in the FE
- Adds the contraints below written in the ticket to the OutputForm in the FE
    - [GP2 Supported] Note: (Blog and GP2 Reports document types - are always set to Yes)
    - [GP2 Supported] Note: This is not required for Training Materials and Procedural Form, do not display this question)
    - [Sharing Status] Note: (Training Materials - always public)
    - [Publish Date] Note: (Publish Date only required if sharing status is Public)

### CODE REVIEW

File: packages/model/src/research-output.ts [hunk 1]
```diff
@@ -182,6 +182,11 @@ export const researchOutputToIdentifierType: Record<
   Report: [],
 };
 
+export const gp2SharingStatuses = ['GP2 Only', 'Public'] as const;
+
+export type GP2ResearchOutputSharingStatus =
+  (typeof gp2SharingStatuses)[number];
+
```
Comment (markcsims) [inline packages/model/src/research-output.ts:189]: This is GP2 specific and should be under the gp2 model. Probably in outputs.

File: packages/gp2-components/src/utils/research-output-form.ts
Comment (markcsims) [inline packages/gp2-components/src/utils/research-output-form.ts:1]: We have outputs in GP2 not research outputs.

Comment (peterstarling) [reply packages/gp2-components/src/utils/research-output-form.ts:1]: and i wonder whether that is a good thing or a bad thing that they are different but only a little different. Like... the idea is to eventually nearly align them two so should we also not name them as research outputs?

File: packages/gp2-components/src/templates/OutputForm.tsx [hunk 2]
```diff
@@ -83,14 +96,36 @@ const OutputForm: React.FC<OutputFormType> = ({
...
+  const [newGp2Supported, setGp2Supported] = useState<
+    gp2Model.DecisionOption | undefined
+  >(isGP2SupportedAlwaysTrue ? 'Yes' : gp2Supported);
+  const [newSharingStatus, setSharingStatus] =
+    useState<gp2Model.OutputSharingStatus>(
+      isAlwaysPublic ? 'Public' : sharingStatus || 'GP2 Only',
+    );
+
+  const [newPublishDate, setPublishDate] = useState<Date | undefined>(
+    getPublishDate(publishDate) || undefined,
```
[the remark this answers is not in the document]
Comment (gabiayako) [reply packages/gp2-components/src/templates/OutputForm.tsx]: done! committed here [review: refactor to remove output-form.ts file](https://github.com/yldio/asap-hub/pull/3713/commits/3537aac09a2915ce95245573908e43f69cf939bb)

**Review by markcsims — APPROVED** [review]

**Pull request merged**